src/components/HistorySideBar.tsx

```diff
@@ -102,11 +106,32 @@ export const HistorySideBar: React.FunctionComponent<IHistorySideBarProps> = (
       }
     };
 
+  /**
+   * Commit info for 'Uncommitted Changes' history.
+   */
+  const uncommitted = React.useMemo<Git.ISingleCommitInfo>(() => {
+    return {
+      author: props.trans.__('You'),
+      // INDEX or WORKING special ref
+      commit: `${+(props.model.selectedHistoryFile?.status === 'staged')}`,
```

Comment:
That is a nice trick but it makes the code less readable. You should always prefer readability (expect when performance is a prime concerned - but even then what seems like a more efficient written code may not be efficient for the processor)

```suggestion
      commit: `${props.model.selectedHistoryFile?.status === 'staged' ? Git.Diff.SpecialRef.INDEX : Git.Diff.SpecialRef.WORKING }`,
```